### ISSUE TITLE
 v8 for .. of statement crash

### ISSUE THREAD

**Issue by koculu** [microsoft/ClearScript#299]
I am getting a crash with v8 for .. of statement on .NET native array in my ASP.NET application.
I tried to reproduce this with a simple console application with no chance.
I hope someone might help by checking exception logs.

Exception logs:
https://gist.github.com/koculu/cfc5b6f4f2cae1fb15e4c77e08b29470

Scenario:
C# side object definition that's being passed to script:

```
/* I have a dynamic object model which contains a pageScripts member as following.  */
 model.pageScripts = new object[]{ "some string" }
```

Javascript for .. of statement on .NET Array that crashes the whole app:
```
for (const script of model.pageScripts) {
}
```

If I replace for .. of statement with standard for loop, script executes fine with no crash.
```
for (let i = 0; i < model.pageScripts.Length ; ++i) {
}
```

Tested Versions: 7.1.4, 7.1.5, 7.1.6, 7.1.7
OS: Windows 10 & Ubuntu 20.04.2 LTS (Docker)

**Comment by ClearScriptLib** [comment]
Hi @koculu,

>I tried to reproduce this with a simple console application with no chance.

Sorry, just to be sure we understand, you're saying that you created a simple console test application, but it failed to reproduce the crash. Is that correct?

>Tested Versions: 7.1.4, 7.1.5, 7.1.6, 7.1.7
OS: Windows 10 & Ubuntu 20.04.2 LTS (Docker)

If we understand correctly, you run your application either (a) natively on Windows, or (b) in a Linux Docker image on Windows. If that's correct, does the crash occur in both environments?

A couple more questions:

1. Does the string array you're iterating tend to be very large, perhaps on the order of millions of elements?
2. Your exception stack is a bit deep – nearly 400 frames – and re-enters the `RenderView` method several times. Does that seem correct? In other words, is `RenderView` designed to be re-entrant?

Thanks!

**Comment by ClearScriptLib** [comment]
Hi @koculu,

Thanks for providing the test case! We'll have a fix in the 7.1.7 final release, which is due any day now.

This bug causes a crash whenever the JavaScript iteration or async iteration protocol is used in a `V8ScriptEngine` instance constructed with `V8ScriptEngineFlags.DisableGlobalMembers`. The workaround is simply to avoid that flag; the GlobalMembers feature no longer incurs a performance penalty anyway.

Thanks for reporting this issue!

**Comment by ClearScriptLib** [comment]
Fixed in [Version 7.1.7](https://github.com/microsoft/ClearScript/releases/tag/7.1.7).
